Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -169,6 +357,7 @@ namespace GradientSignal
         // If the preview shouldn't be active, use an invalid entityId
         m_gradientEntityId = AZ::EntityId();
 
+        AZ::TickBus::Handler::BusDisconnect();
```

Comment:
Deactivate should also cancel the job and block waiting for it to complete.

Reply:
Good call, I've implemented this functionality now